### DISCUSSION TITLE
[file-service] Pre-Signed url generation

### DISCUSSION THREAD

**Discussion by fabio-percivaldi** [mia-platform/community#68] [Ideas]
it could be usefull to have an endpoint to generate pre-signed url for aws/gcp files.

**Comment by davidebianchi** [comment]
Do you mean this [signed url for GCP](https://cloud.google.com/storage/docs/access-control/signed-urls) and [presigned url for AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-presigned-url.html#PresignedUrlUploadObject-LimitCapabilities), correct?

Can you explain the use case to use this url?

**Reply by fabio-percivaldi** [reply]
yes exactly, we would like to be able to upload files internally, but give the possibility to share the uploaded files with users that will download the files from the public internet that would not like to use APIs

**Reply by davidebianchi** [reply]
Ok, thanks. Can you share how the feature would be? There will be an expiration time to propose I suppose, and it should be configurable.

We should also document when it is possible to use it, for example if the storage is not exposed to the public internet, it could not be possible to use the presigned url from the browser.
